### ISSUE TITLE
publisher_acl with regex on username not working and has no documentation

### ISSUE THREAD

**Issue by UtahDave** [saltstack/salt#43124]
### Description of Issue/Question
This issue: https://github.com/saltstack/salt/issues/23194 asked for regex on the username line in `publisher_acl`

@DmitryKuzmenko added support for this here: https://github.com/saltstack/salt/pull/34268

I can find no documentation on how this is supposed to work other than this comment here: https://github.com/saltstack/salt/issues/23194#issuecomment-267570004

I've tested on both `2016.11.7` and `2017.7.1` and get the same error.

My master config only has the following lines:

```
publisher_acl:
  (user1|user2):
    - .*
```

When I attempt to start up the salt-master I get the following error:

```
[root@issue1699 ~]# salt-master -l debug
[DEBUG   ] Reading configuration from /etc/salt/master
[DEBUG   ] Using cached minion ID from /etc/salt/minion_id: issue1699
[DEBUG   ] Configuration file path: /etc/salt/master
[WARNING ] Insecure logging configuration detected! Sensitive data may be logged.
[INFO    ] Setting up the Salt Master
[DEBUG   ] Loaded master key: /etc/salt/pki/master/master.pem
[PROFILE ] Beginning pwd.getpwall() call in masterarpi access_keys function
[PROFILE ] End pwd.getpwall() call in masterarpi access_keys function
[INFO    ] Preparing the root key for local communication
[DEBUG   ] Removing stale keyfile: /var/cache/salt/master/.root_key
[INFO    ] Preparing the (user1|user2) key for local communication
[DEBUG   ] Removing stale keyfile: /var/cache/salt/master/.(user1|user2)_key
[ERROR   ] An un-handled exception was caught by salt's global exception handler:
KeyError: 'getpwnam(): name not found: (user1|user2)'
Traceback (most recent call last):
  File "/usr/bin/salt-master", line 22, in <module>
    salt_master()
  File "/usr/lib/python2.7/site-packages/salt/scripts.py", line 92, in salt_master
    master.start()
  File "/usr/lib/python2.7/site-packages/salt/cli/daemons.py", line 204, in start
    super(Master, self).start()
  File "/usr/lib/python2.7/site-packages/salt/utils/parsers.py", line 1032, in start
    self.prepare()
  File "/usr/lib/python2.7/site-packages/salt/cli/daemons.py", line 185, in prepare
    self.master = salt.master.Master(self.config)
  File "/usr/lib/python2.7/site-packages/salt/master.py", line 395, in __init__
    SMaster.__init__(self, opts)
  File "/usr/lib/python2.7/site-packages/salt/master.py", line 120, in __init__
    self.key = self.__prep_key()
  File "/usr/lib/python2.7/site-packages/salt/master.py", line 144, in __prep_key
    return salt.daemons.masterapi.access_keys(self.opts)
  File "/usr/lib/python2.7/site-packages/salt/daemons/masterapi.py", line 265, in access_keys
    keys[user] = mk_key(opts, user)
  File "/usr/lib/python2.7/site-packages/salt/daemons/masterapi.py", line 236, in mk_key
    os.chown(keyfile, pwd.getpwnam(user).pw_uid, -1)
KeyError: 'getpwnam(): name not found: (user1|user2)'
Traceback (most recent call last):
  File "/usr/bin/salt-master", line 22, in <module>
    salt_master()
  File "/usr/lib/python2.7/site-packages/salt/scripts.py", line 92, in salt_master
    master.start()
  File "/usr/lib/python2.7/site-packages/salt/cli/daemons.py", line 204, in start
    super(Master, self).start()
  File "/usr/lib/python2.7/site-packages/salt/utils/parsers.py", line 1032, in start
    self.prepare()
  File "/usr/lib/python2.7/site-packages/salt/cli/daemons.py", line 185, in prepare
    self.master = salt.master.Master(self.config)
  File "/usr/lib/python2.7/site-packages/salt/master.py", line 395, in __init__
    SMaster.__init__(self, opts)
  File "/usr/lib/python2.7/site-packages/salt/master.py", line 120, in __init__
    self.key = self.__prep_key()
  File "/usr/lib/python2.7/site-packages/salt/master.py", line 144, in __prep_key
    return salt.daemons.masterapi.access_keys(self.opts)
  File "/usr/lib/python2.7/site-packages/salt/daemons/masterapi.py", line 265, in access_keys
    keys[user] = mk_key(opts, user)
  File "/usr/lib/python2.7/site-packages/salt/daemons/masterapi.py", line 236, in mk_key
    os.chown(keyfile, pwd.getpwnam(user).pw_uid, -1)
KeyError: 'getpwnam(): name not found: (user1|user2)'
```
And the salt-master shuts down.

If I change the master config to the following, then both `user1` and `user2` can run commands successfully.

```
publisher_acl:
  user1:
    - .*
  user2:
    - .*
```

**Comment by UtahDave** [comment]
ZD 1699

**Comment by UtahDave** [comment]
If this feature is supposed to work I think this page should be updated with some examples using regex for the username line:  https://docs.saltstack.com/en/latest/ref/publisheracl.html

**Comment by DmitryKuzmenko** [comment]
@UtahDave you're right. I've updated all the logic related to runtime patterns support in `publisher_acl` but I've missed the initialization part that generates the users keys. #43467 fixes all problems I've found there. Also I've added an example section to the docs as you've proposed.

The PR is targeted to 2017.7 branch. Let me know if there's something wrong.

**Comment by cachedout** [comment]
CLosed via #43467
